hibernate-core/src/main/java/org/hibernate/boot/beanvalidation/TypeSafeActivator.java

```diff
@@ -72,6 +75,9 @@ class TypeSafeActivator {
 
 	private static final CoreMessageLogger LOG = Logger.getMessageLogger( MethodHandles.lookup(), CoreMessageLogger.class, TypeSafeActivator.class.getName() );
 
+	private static final Map<Class<? extends Annotation>, Boolean>
```

Comment:
Storing class references to application classes in a static field is unfortunately going to be a problem, unless you clean up the storage properly. Also, concurrent deployments could potentially run into race conditions, so I would suggest you use a `ThreadLocal<Map<Class<? extends Annotation>, Boolean>>` instead and also make sure you clear it in a finally block within the `activate` method.